esmf_regrid/experimental/io.py

```diff
@@ -43,6 +57,49 @@ def _add_mask_to_cube(mask, cube, name):
         cube.add_aux_coord(mask_coord, list(range(cube.ndim)))
 
 
+@contextmanager
+def _managed_var_name(src_cube, tgt_cube):
```

Comment:
I don't really see the purpose of defining this as a _context manager_, 
since it anyway only makes some _permanent_ changes to the arguments (mutable)?
So, it has no 'yield', and no code to "undo afterwards".

So, instead of the style ..
```
with change_cubes(cubes):
   operate(cubes)
```
You could simply
```
change_cubes(cubes)
operate(cubes)
```

Reply:
The "undo afterwards" code should be in the `finally:` part. I've added some tests to show that this does actually revert back the changed to the original cube. See https://github.com/SciTools-incubator/iris-esmf-regrid/pull/357/commits/05209672d69270925eacbe4a7772d05288d07adf

Reply:
OK I clearly hadn't read this carefully enough.  There **_is_** a `yield` (!)